test/test_torch.py

```diff
@@ -7745,40 +7745,57 @@ def cholesky_test_helper(n, batchsize, device, upper):
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    @dtypes(torch.double)
+    @dtypes(torch.float, torch.double, torch.complex64, torch.complex128)
     def test_cholesky_batched(self, device, dtype):
         from torch.testing._internal.common_utils import random_symmetric_pd_matrix
 
         def cholesky_test_helper(n, batch_dims, upper):
-            A = random_symmetric_pd_matrix(n, *batch_dims, dtype=dtype, device=device)
+            # This is a workaround while there is no support for complex random_symmetric_pd_matrix
+            if dtype.is_complex:
+                real_dtype = torch.float32 if dtype is torch.complex64 else torch.float64
+                A_real = random_symmetric_pd_matrix(n, *batch_dims, dtype=real_dtype, device=device)
+                A_imag = random_symmetric_pd_matrix(n, *batch_dims, dtype=real_dtype, device=device)
+                A = A_real + 1j * A_imag
+                A = 0.5 * (A + A.transpose(-2, -1).conj())
```

Comment:
This makes imaginary part 0 (A is symmetric matrix, A.transpose(-2,-1) is equal to A, A+A.conj() kills imaginary part), so the tests aren't testing much. You want hermitian positive definite input.

Reply:
I've fixed this.